ROS2 port

nkhedekar opened the pull request:
Fixes #4

SteveMacenski commented on open3d_conversions/package.xml:
```diff
@@ -12,15 +13,12 @@
-  <depend>roscpp</depend>
+  <depend>rclcpp</depend>
   <depend>sensor_msgs</depend>
-  <depend>eigen</depend>
```
You still use Eigen3, needs to have a depend

nkhedekar replied:
Fixed in d09626d

SteveMacenski reviewed (commented):
LGTM. I didn't look at the code changes in the file since I assumed you know what you're doing ;-) 

Other things you should consider is enabling the automatic linters and linting the code. I see a bunch of uncrustify and cpplint violations.

nkhedekar commented:
Uncrustify seems to be fine: 

```bash
$ ament_uncrustify .
No code style divergence in file 'perception_open3d/open3d_conversions/include/open3d_conversions/open3d_conversions.hpp'

No code style divergence in file 'perception_open3d/open3d_conversions/include/open3d_conversions/visibility_control.h'

No code style divergence in file 'perception_open3d/open3d_conversions/src/open3d_conversions.cpp'

No code style divergence in file 'perception_open3d/open3d_conversions/test/test_open3d_conversions.cpp'

No problems found

```

SteveMacenski reviewed (approved):
Try running with all the linters, the autolint ament macro will run this. I see a white space (at minimum in package.xml)

nkhedekar commented:
- Removed extra spaces
- Removed unused variables (using `-Wall -Wextra -Wpedantic`)
- Checked with `ament_cpplint`:
```bash
$ ament_cpplint .
Using '--root=/home/nkhedekar/Desktop/workspaces/o3d_ws/src/perception_open3d/open3d_conversions/include' argument

Done processing /home/nkhedekar/Desktop/workspaces/o3d_ws/src/perception_open3d/open3d_conversions/include/open3d_conversions/open3d_conversions.hpp

Done processing /home/nkhedekar/Desktop/workspaces/o3d_ws/src/perception_open3d/open3d_conversions/include/open3d_conversions/visibility_control.h

Using '--root=/home/nkhedekar/Desktop/workspaces/o3d_ws/src/perception_open3d/open3d_conversions/src' argument

Done processing /home/nkhedekar/Desktop/workspaces/o3d_ws/src/perception_open3d/open3d_conversions/src/open3d_conversions.cpp

Using '--root=/home/nkhedekar/Desktop/workspaces/o3d_ws/src/perception_open3d/open3d_conversions/test' argument

Done processing /home/nkhedekar/Desktop/workspaces/o3d_ws/src/perception_open3d/open3d_conversions/test/test_open3d_conversions.cpp

No problems found
```
- Checked with `ament_cppcheck`

```bash
$ ament_cppcheck .
No problems found
```

- Checked with `ament_uncrustify`

```bash
$ ament_uncrustify .
No code style divergence in file 'include/open3d_conversions/open3d_conversions.hpp'

No code style divergence in file 'include/open3d_conversions/visibility_control.h'

No code style divergence in file 'src/open3d_conversions.cpp'

No code style divergence in file 'test/test_open3d_conversions.cpp'

No problems found
```
